REVIEW_GUIDELINES.md

```diff
@@ -0,0 +1,63 @@
+------------------------
+Approximately every two weeks, the ROS Boss of the corresponding ROS distribution will do a sync, which updates existing packages and releases new packages of the ROS distribution to the general public.
+Prior to the sync, the ROS Boss will announce a "sync freeze" on [Discourse](https://discourse.ros.org/).
+During the sync freeze, no new packages or package updates will be merged into that particular ROS distribution without the express consent of the ROS Boss.
+Every ROS distribution is on a different sync schedule.
+
+Types of Pull Requests
+----------------------
+There are generally 5 different types of pull requests that are opened against this repository (these are in order of most likely to least likely to happen):
+
+1.  A version update to an existing package in a ROS distribution.  An example of this kind of PR is [25822](https://github.com/ros/rosdistro/pull/25822).  In these types of pull requests, the package owner is updating a major, minor, patch, or debinc number in the package.  As long as the ROS distribution isn’t in a “sync freeze”, these PRs will be merged without further review.  If the ROS distribution is in a "sync freeze", the PR will be approved with a note saying something like “Holding for sync freeze” with a link to the relevant discourse post. Once the distribution is out of sync freeze, these PRs will then be merged.
+
+1.  A new binary package in a ROS distribution.  An example of this kind of PR is [25857](https://github.com/ros/rosdistro/pull/25857).  An entry is considered a new binary package if it has a `release` field.  These PRs should generally have been opened via bloom; if they aren’t, the reviewer will ask why the submitter hasn't used bloom.  There are two sub-categories of this kind of PR:
+    1.  If the package has been released into a prior ROS distribution, then the reviewer will ensure that the URLs to the doc, release, and source entries are the same as in the previous releases (this will be checked either by looking for the package in a previous distros distribution.yaml file, or by looking at the release repository for a branch named release/{distro}).  If they are the same, and the ROS distribution isn’t in a "sync freeze", then the PR will be merged without further review.  If the URLs are different, then the reviewer will ask the submitter why the URLs are different and whether it can be combined into the original one.
+    1.  For packages that are brand-new to the ROS ecosystem, some due diligence will be done on the source and release repositories.  The source repository should have a top-level LICENSE file, and the license should be one of the open-source ones listed in [https://opensource.org/licenses](https://opensource.org/licenses) (preferably one of the "Popular licenses").  That license should also be reflected in the package.xml file of all sub-packages in the repository.  Both the source and release repositories should be publicly accessible.
```

Comment:
> @clalancette while on call, I run into packages already released that didn't have a top-level LICENSE because the repository held packages with different licenses. What's expected in such cases?

I'm not sure what the official stance is.  For packages that have been in the ROS ecosystem a long time, I typically merge them in whether they have a LICENSE file or not.  Encouraging the submitter to add a LICENSE is appropriate, but not a blocker in my opinion.

Reply:
I think that we could encourage contributors to add `LICENSE.A`, `LICENSE.B` and a `LICENSE` file.
Where for example: A="BSD", B="Apache".
The `LICENSE` file should explain which packages on the repo are licensed under which license, and link to the specific files.

A lot of dual licensed repositories do that (dual licensing isn't the same, but as long as the explanation in the `LICENSE` is correct, I think it's applicable).

Reply:
For multi-license repositories, sure, that makes some sense.  For repositories that have just a single license (which I would say is most of them), I think a single LICENSE file is probably sufficient.  It's already a burden that a lot of developers don't want to have to deal with; I would prefer to avoid making it even more onerous.

Reply:
> I think a single LICENSE file is probably sufficient. It's already a burden that a lot of developers don't want to have to deal with; I would prefer to avoid making it even more onerous.

Yeah, for repositories that have a single license for sure.
But I think that it's still good to encourage people in the community to create a top level license file, even if the code in the repository has different licenses or is multi-licensed.

Maybe that shouldn't be a blocker to merger rosdistro PRs though.

Reply:
that sounds good to me